### ISSUE TITLE
Document license under which audioop is used

### ISSUE THREAD

**Issue by brettcannon** [python/cpython#46968]
BPO | [2716](https://bugs.python.org/issue2716)
--- | :---
Nosy | @akuchling, @birkenfeld, @terryjreedy, @vstinner, @benjaminp, @VanL, @furkanonder
PRs | <li>python/cpython#19972</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2008-04-29.00:22:49.758>
labels = ['3.8', '3.7', 'library', '3.9', 'docs']
title = 'Document license under which audioop is used'
updated_at = <Date 2020-05-07.00:56:43.829>
user = 'https://github.com/brettcannon'
```

bugs.python.org fields:
```python
activity = <Date 2020-05-07.00:56:43.829>
actor = 'terry.reedy'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation', 'Library (Lib)']
creation = <Date 2008-04-29.00:22:49.758>
creator = 'brett.cannon'
dependencies = []
files = []
hgrepos = []
issue_num = 2716
keywords = ['patch']
message_count = 9.0
messages = ['65948', '65952', '66023', '105618', '112507', '170115', '217675', '241760', '368316']
nosy_count = 9.0
nosy_names = ['akuchling', 'georg.brandl', 'terry.reedy', 'vstinner', 'benjamin.peterson', 'vanl', 'docs@python', 'hac.man', 'furkanonder']
pr_nums = ['19972']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue2716'
versions = ['Python 3.7', 'Python 3.8', 'Python 3.9']
```

</p></details>

**Comment by brettcannon** [comment]
- main: https://github.com/python/cpython/pull/19972
- 3.11: https://github.com/python/cpython/pull/98531
- 3.10: https://github.com/python/cpython/pull/98532

**Comment by ambv** [comment]
This library is removed from Python 3.13. However, the currently used license file in 3.10 - 3.12 is invalid and should be replaced.

Analysis by Aaron K. Williamson advises us to revert to the Sun Microsystems license text as present in g771.c found in the tarball of sox 12.17.7.

### Full context from Aaron

**April 11, 2022:** Van [comments](https://github.com/python/cpython/pull/19972#issuecomment-1095622743) on the commit saying the license "does not match the declared license from the source file (GPL, or possible LGPL later)," "looks incorrect," and that the correct license "would significantly affect the licensing of Python overall." It's clear from Van's comment that the "source file" he's referring to is the one at the URL in the commit: https://sourceforge.net/projects/sox/files/sox/12.17.7/. If you download the binary (.zip) distribution of the SoX project from that URL, the only declared license is the GPL. According to Van's comment, he was unable to access the "sourcearchive on bpo" because the link was down. So he looked at the source URL identified in the commit and concluded the license was GPL. If he had looked at the [other download](https://sourceforge.net/projects/sox/files/sox/12.17.7/sox-12.17.7.tar.gz/download) (the tar.gz) at that [SourceForge URL](https://sourceforge.net/projects/sox/files/sox/12.17.7/), he would have found the source code for SoX v. 12.17.7. And in that source code distribution, the file g711.c is licensed under the Sun Microsystems license included in the initial commit (which is fully compatible with the PSF License).

**July 30, 2022:** furkanonder makes [another commit](https://github.com/python/cpython/pull/19972/commits/ccec2e71e944c07f45bdbf4f73cd00d77a37b086) replacing the Sun Microsystems license header with the current license header (beginning "Programming the AdLib/Sound Blaster FM Music Chips Version 2.0").

**October 3, 2022:** furkanonder [responds](https://github.com/python/cpython/pull/19972#issuecomment-1265951927) to Van's comment, saying "I have added the license in the source you mentioned." It is unclear why furkanonder thought he was doing what Van asked. The license he added appears to be from a short webpage about programming the Sound Blaster: http://www.shipbrook.net/jeff/sb.html. The license text is not found in the source archive Van linked to -- the license for g711.c in that source archive is the same Sun Microsystems license that was included in the initial commit. Maybe it was just copypasta, but the license in master is not the correct license.

**In conclusion:** just revert the file header to the Sun Microsystems license in the initial commit and everything will be fixed.

**Comment by ambv** [comment]
Order restored to 3.10 - 3.12.
